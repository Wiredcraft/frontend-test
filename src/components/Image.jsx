import React, { useState } from 'react'

export default function Image(props) {

  const { src } = props
  const [pulse, setPulse] = useState(true)

  const getHeight = (e) => {
    const path = src.split('?')[0]
    return path.slice(path.length - 3, path.length)
  }

	return (
		<div className={`image-wrapper ${pulse ? 'pulse' : ''}`}>
			<img
        className="image"
        alt={src}
        style={{ height: `${getHeight()}px` }}
        onLoad={() => setPulse(false)}
      />
		</div>
	)
}
