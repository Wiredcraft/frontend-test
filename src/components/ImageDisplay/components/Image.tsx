import React from 'react'
import { ImageProps } from './interfaces'

const Image: React.FC<ImageProps> = (props) => {
  return <img {...props} />
}

export default Image
