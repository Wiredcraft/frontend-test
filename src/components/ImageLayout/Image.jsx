import React, { useState } from 'react';
import cx from 'classnames';
import styles from './Image.module.less';

function Image (props) {
  const {
    src,
    className,
    top,
    left,
    width,
    height
  } = props;

  const [showImg, setShowImg] = useState(false);

  const style = {
    top,
    left,
    width,
    height
  };
  return (
    <div className={styles.imageContainer} style={style}>
      <img
        src={src}
        onLoad={() => setShowImg(true)}
        className={cx({
          [styles.image]: true, 
          [styles.hidden]: !showImg,
          [className]: !!className
        })}
      />
     {!showImg && <div className={styles.placeholder} style={style}></div>}
    </div>
  )
  
}


export default Image;