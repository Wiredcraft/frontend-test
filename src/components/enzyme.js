import 'regenerator-runtime/runtime';
import Enzyme, { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
export { shallow, mount, render };
export default Enzyme;
