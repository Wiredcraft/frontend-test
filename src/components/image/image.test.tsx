/**
 * Image component test
 * Fri Jul 24 15:18:27 2020
 * by xiaoT
 */
import React, { FC } from 'react'
import ReactDom from 'react-dom'
import { shallow, mount } from 'enzyme'

import Image from './'

const testPorps = {
  placeholder: 'https://gblobscdn.gitbook.com/spaces%2F-LceGMK-Zxa6_-QeGdy1%2Favatar.png?alt=media',
  src: 'https://picsum.photos/240/379?random=371',
  alt: 'image name',
  style: {
    backgroudColor: '#ccc'
  }
}

describe('Image Test', () => {
  let imageComponent: any
  let imageEle: any

  beforeEach(() => {
    imageComponent  = shallow(<Image {...testPorps} />)
    imageEle = imageComponent.find('img')
  })
  afterEach(() => {
    imageComponent.unmount()
    imageEle = null
  })
  // should be match the snapshot
  it('should be match the snapshot', () => {
    expect(imageComponent.html()).toMatchSnapshot()
  })
  // Image should be have an image element
  it('should be have an image element', () => {
    expect(imageEle.length).toEqual(1)
  })
  // Image should be have a src attribute
  it('should be have a src attribute', () => {
    const src = imageEle.prop('src')
    expect(src).toEqual(testPorps.src)
  })
  // Image alt attribute should be equal testPorps.alt
  it('Image alt attribute should be equal testPorps.alt', () => {
    const alt = imageEle.prop('alt')
    expect(alt).toEqual(testPorps.alt)
  })
  // image container style attribute should be equal testPorps.style
  it('Image style attribute should be equal testPorps.style', () => {
    const containerStyle = imageComponent.prop('style')
    expect(containerStyle).toHaveProperty('backgroudColor', '#ccc')
  })
  // Image should be have a data-lazy attribute
  it('should be have a data-lazy attribute', () => {
    const dataLazy = imageEle.prop('data-lazy')
    expect(dataLazy).toEqual(testPorps.placeholder)
  })
})