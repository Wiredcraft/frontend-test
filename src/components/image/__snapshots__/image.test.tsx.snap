// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Test should be match the snapshot 1`] = `"<div class=\\"image-container\\"><img data-lazy=\\"https://gblobscdn.gitbook.com/spaces%2F-LceGMK-Zxa6_-QeGdy1%2Favatar.png?alt=media\\" src=\\"https://picsum.photos/240/379?random=371\\" alt=\\"\\"/></div>"`;
